src/prefect/infrastructure/provisioners/cloud_run.py

```diff
@@ -168,15 +191,21 @@ async def _create_gcp_credentials_block(
             )
             return block_doc.id
 
-    async def provision(self, work_pool_name: str, base_job_template: dict):
-        async with get_client() as client:
-            await self._check_for_gcloud()
-            self._project = await self._get_current_project()
-            self._region = await self._get_default_region()
-
-            table = Panel(
-                dedent(
-                    f"""\
+    @inject_client
+    async def provision(
+        self,
+        work_pool_name: str,
+        base_job_template: dict,
+        client: Optional[PrefectClient] = None,
+    ) -> Dict[str, Any]:
+        assert client, "Client injection failed"
+        await self._check_for_gcloud()
+        self._project = await self._get_project()
+        self._region = await self._get_default_region()
+
+        table = Panel(
+            dedent(
+                f"""\
                     Here's what we'll do to provision infrastructure for this work pool:
 
                         Updates in GCP project [blue]{self._project}[/] in region [blue]{self._region}[/]
```

Comment:
Q: Should we say "Here's what we'll do" versus "Here's what Prefect will do"? I'm unsure of which voice and tone makes sense.

Reply:
Or maybe we don't make many references to agency/responsibility so perhaps there isn't a good precedent.

Reply:
I updated to `Here are the updates necessary to provision infrastructure for your work pool {work_pool_name}`. Let me know if that's an improvement.

Reply:
What about like:

`Provisioning infrastructure for this work pool will require:`

since that flows well with the bullet points (which both start with "updates in GCP/Prefect")